pennylane/ops/qubit.py

```diff
@@ -853,6 +975,18 @@ class QubitUnitary(Operation):
     par_domain = "A"
     grad_method = None
 
+    @staticmethod
+    def _matrix(*params):
+        U = np.asarray(params[0])
+
+        if U.shape[0] != U.shape[1]:
+            raise ValueError("Operator must be a square matrix.")
```

Comment:
For this check, if a square matrix is passed with a shape ``(dim, dim, 1)``, then this check will pass, however, the  ``np.allclose(U @ U.conj().T`` part in the next fails. Would it be perhaps worth reshaping here, or somehow checking further for more than the first two indices of ``shape``?

An examle would be passing `` [[[1], [0]], [[0], [1]]]`` as ``params[0]``.

Reply:
Is this a use case that is likely to happen? I'm hesitant to add more input validation, unless it is something that could happen accidentally or regularly (and I don't think I've ever seen that particular input occur).

Reply:
These cases appeared in certain tests when using the ``apply`` method, where lists containing exactly one operator appeared. I've mostly covered transforming these tests in the related PR, so apart from these there might not be many cases.